app/web_framework/cluster_base_handler.py

```diff
@@ -46,6 +56,12 @@ def _handle_request_exception(self, ex):
         self.set_status(status_code)
         self.finish()
 
+    def on_finish(self):
+        if self._route_node is not None:
+            http_request_duration_seconds.labels(self.request.method,  # pylint: disable=no-member
+                                                 self._route_node.regex(),  # pylint: disable=no-member
```

Comment:
You might not need the `disable=no-member` now. I wonder if you were seeing an error on `self.route` before because that legitimately did not exist.